Dockerfile

```diff
@@ -0,0 +1,35 @@
+from python:3.8.2-slim-buster as base
+
+ENV APP_INSTALL=/app
+ENV POETRY_HOME=/poetry
+ENV PATH=${POETRY_HOME}/bin:${PATH}
+ENV PYTHON_PATH=${APP_INSTALL}
+
+# Install curl
+RUN apt-get update && apt-get install curl -y
+
+# Install poetry 
+RUN curl -sSL https://raw.githubusercontent.com/python-poetry/poetry/master/get-poetry.py | python
+
+
+# copy across app code
+WORKDIR ${APP_INSTALL}
+COPY . /app
+RUN poetry install
+
+# list all copied files
+RUN ls -la ./
+
+
+# define entry point , and default launch command
+
+# for development
+FROM base as development
+EXPOSE 5002
```

Comment:
This is wrong. This command means that you expect the **container** to have something running on port 5002, but you don't. `flask run` will start a server listening to port 5000 by default.

The image doesn't specify what port you use on your machine (the host machine) because it can't know. That's determined entirely by what you put in the `docker run` command.

Similarly the production image's expose is also wrong. It should be `EXPOSE 8000` because that's what gunicorn will run on (you could specify a different port with e.g. `--bind 0.0.0.0:8080`)